TheMET/TheMET/ViewControllers/CatalogViewController.swift

```diff
@@ -21,20 +39,132 @@ class CatalogViewController: UIViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
         self.navigationController?.navigationBar.standardAppearance = self.navigationItem.apply(title: NSLocalizedString("catalog_screen_title", comment: ""), color: UIColor(named: "plum"), fontName: NSLocalizedString("serif_font", comment: ""), fontSize: 22)
-        let  newView: TagListView = TagListView()
-//        let newView = CatalogContentView.constructView()
-//        newView.tags = [ "Винсент Ван Гог", "Живопись", "Арт", "Европа","Винсент Ван Гог", "Живопись", "Арт", "Европа", "Арт", "Европа"]
-//        newView.translatesAutoresizingMaskIntoConstraints = false
-        newView.tags = [ "Vincent Van Goghl", "Painting","Vincent", "Painting", "Dutch", "Vincent Van Gogh", "Painting", "Vincent Van Gogh", "Vincent Van Goghl", "Painting","Vincent", "Painting", "Dutch", "Vincent Van Gogh", "Painting", "Vincent Van Gogh", "Vincent Van Goghl", "Painting","Vincent", "Painting", "Dutch", "Vincent Van Gogh", "Painting", "Vincent Van Gogh"]
-        newView.translatesAutoresizingMaskIntoConstraints = false
-        self.view.addSubview(newView)
-        NSLayoutConstraint.activate([
-            newView.leadingAnchor.constraint(equalTo: self.view.leadingAnchor, constant: 10),
-            newView.trailingAnchor.constraint(equalTo: self.view.trailingAnchor, constant: -10),
-            newView.topAnchor.constraint(equalTo: self.view.topAnchor, constant: 150),
-//            newView.bottomAnchor.constraint(equalTo: self.view.bottomAnchor, constant: 0)
-        ])
+        
+        loadingCatalogView.translatesAutoresizingMaskIntoConstraints = false
+        failedCatalogView.translatesAutoresizingMaskIntoConstraints = false
+        loadedCatalogView.translatesAutoresizingMaskIntoConstraints = false
+        self.updateContent()
+        self.add(catalogSubview: failedCatalogView, stretchView: true)
+        self.add(catalogSubview: loadedCatalogView, stretchView: false)
+        self.add(catalogSubview: loadingCatalogView, stretchView: true)
+        self.loadCatalogCellDataList { catalogCellDataList in
+            guard let catalogCellDataList = catalogCellDataList else {
+                self.contentStatus = .failed
+                return
+            }
+            self.contentStatus = .loaded(catalogCellDataList)
+        }
+        self.updateContent()
```

Comment:
You're executing `self.updateContent()` twice:line 46 and line 57

Reply:
fixed